lib/postgres/scan.go

```
@@ -105,16 +103,37 @@ func scanTableQuery(args scanTableQueryArgs) (string, error) {
 	), nil
 }
 
-func shouldQuoteValue(col schema.Column) bool {
-	kindDetail := pgDebezium.ColumnToField(col).ToKindDetails()
-	switch kindDetail.Kind {
-	case typing.Invalid.Kind:
-		slog.Warn("Could not determine data type for column", slog.String("colName", col.Name))
-		return true
-	case typing.String.Kind, typing.Struct.Kind, typing.ETime.Kind, typing.EDecimal.Kind:
-		return true
+func shouldQuoteValue(col schema.Column) (bool, error) {
+	switch col.Type {
+	case schema.InvalidDataType:
+		return false, fmt.Errorf("invalid data type")
```

Expected?

---

I prefer this actually, it'll flag unknown data types as opposed to failing silently.

---

Yep, we should never be calling this function with an `InvalidDataType`.

---

When we load the columns we throw an error if we cannot determine any of the types: https://github.com/artie-labs/reader/blob/ed33aac4f28361ed59b28f6e3f0426cf6aad3d0f/lib/postgres/schema/schema.go#L96